static/app/views/admin/adminRelays.tsx

```diff
@@ -52,14 +53,14 @@ class AdminRelays extends Component<Props, State> {
       </td>,
       <td key="tools" style={{textAlign: 'right'}}>
         <span className="editor-tools">
-          <LinkWithConfirmation
-            className="danger"
-            title="Remove"
+          <Confirm
```

Comment:
side-note: I read through the `Confirm` component and I’m not a fan of doing implicit children cloning. It just puts an `onClick` on `children`, so it won’t work if the direct child doesn’t support `onClick` or if there are multiple children in a `Fragment`. It also doesn’t merge with existing `onClick` props on the child - it will just overwrite them!

If we want this component, is should be with children-as-a-function only that provides those props, which can then be forwarded to  / merged with in a type-safe way to components that support this prop.

Otherwise, as `useConfirm()` hook that returns props is a good alternative.

Reply:
Agree, I can tackle this in a separate PR, but I think the hook that returns onClick is the right thing here.

Reply:
I should add that the implicit cloning has also lead to a few nasty bugs in the past where components like confirm were swallowing the onClick prop on the provided child